vllm/v1/engine/core.py

```diff
@@ -101,6 +131,9 @@ def add_request(self, request: EngineCoreRequest):
         # take 10-50 ms, which can cause a spike in the latency. We should
         # consider moving this to a separate thread.
         if req.mm_data:
+
```

Comment:
Thoughts on doing this on the frontend engine process (i.e. `v1/engine/processor.py::Processor`) before sending to the EngineCore? 

IIUC: this `add_request` is called on the EngineCore process, meaning it's sync blocking the model executor too?

Reply:
Yea this is already planned. Eventually the engine will live on the frontend, together with input processor. https://github.com/vllm-project/vllm/pull/10044 is working towards this direction.

Reply:
@rickyyx I think it is a good idea, I can try it.